### PR TITLE
feat(comment): delete comments implemention

### LINKED ISSUE #1179: Add “Delete” kebab menu option to comments
Part of #654

### LINKED ISSUE #1180: Implement UX workflow for comment deletion
Part of #654

### LINKED ISSUE #1181: Integrate API to send comment deletion request
Part of #654

### PR DESCRIPTION
**Pull request by debloper**
- Kebab menu added to the comments by the logged in user
  - Has Edit and Delete dropdown menu
  - Opens the comment in edit mode, on activation of Edit menu button
  - Delete button opens modal for delete confirmation
- Affirming on confirmation modal sends delete request to API
- post-delete update local collection of comments to reflect the changes
- **Important:** the modal backdrop behavior is weird (because of messed up z-indexes in many places), it has been handled with minimal collateral impact, may not reflect exactly as UX suggestion.
- Resolved the kebab menu alignment issue as per the UX suggestion
- Fixed the tests by including modal-module in the specs

Closes #1179, closes #1180, closes #1181, part of story #654

### CODE REVIEW

File: src/app/work-item/work-item-detail/work-item-comment/work-item-comment.component.html [hunk 1]
```diff
@@ -40,8 +40,7 @@ <h5 class="comment-authr-name">
                 </h5>
 
                 <small class="createTimeBlock" id="{{'comment_time_' + counter}}">
```
Comment (mindreeper2420) [inline src/app/work-item/work-item-detail/work-item-comment/work-item-comment.component.html]: this should just have the timestamp, aligned to the right of the comment block (taking into account the kebab menu, when applicable).

Comment (debloper) [inline src/app/work-item/work-item-detail/work-item-comment/work-item-comment.component.html]: Resolved, please update review.

**Comment by mindreeper2420** [comment]
When I run this locally, I just see an empty space where the comment should be. Can you attach a screenshot?

screen shot 2017-03-30 at 1 54 54 pm

**Comment by debloper** [comment]
> When I run this locally, I just see an empty space where the comment should be. Can you attach a screenshot?
Does your workitem have any comment? @mindreeper2420

**Comment by mindreeper2420** [comment]
@nimishamukherjee Delete in the modal confirmation should be red.

**Comment by mindreeper2420** [comment]
@debloper my work items did not. When I tried to add a comment, it failed. As I cannot test for an unknown reason and two others approved, I will approve and look after it has been merged.

**Comment by debloper** [comment]
@nimishamukherjee as @mindreeper2420 confirms, the delete is a (pseudo)destructive operation, as such it's better have a visual indication for that (and the danger button is PF way to do it).

@mindreeper2420 I'm unsure about your setup then, whether everything is alright. I know some stuff are regressing for the recent observable changes, so if you happen to find out what's stopping you from adding the comments, please do report issue. If you get a particular error message or HTTP code, that'll help.

**Pull request merged**